docs/Developer Documentation/Security Manager Design/index.md

```diff
@@ -0,0 +1,74 @@
+# Security Manager Guide
+
+### What is the Security Manager?
+
+Secure communication in SDL Core is provided by the Security Manager interface, implemented in Core as the [SecurityManagerImpl](https://github.com/smartdevicelink/sdl_core/blob/master/src/components/security_manager/src/security_manager_impl.cc).
+
+Under the hood the Security Manager uses OpenSSL to to complete handshakes and encrypt/decrypt data. All OpenSSL operations are abstracted away by the [Crypto Manager](https://github.com/smartdevicelink/sdl_core/blob/master/src/components/security_manager/src/crypto_manager_impl.cc) which provides a factory for SSL Context objects, the ability to update certificates from Certificate Authorities and SSL error reporting. The SSL Context objects created by the Crypto Manager can be used to establish SSL connections, as well as to encrypt or decrypt data. Events within the Security Manager will be propagated to any Security Manager Listeners that register with the Security Manager. The Protocol Handler implementation in Core implements the SecurityManagerListener interface in order to send protocol layer responses to handshakes.
+
+### What events are generated?
+
+#### OnHandshakeDone
+
+When a handshake has been completed, the `OnHandshakeDone` is called on each registered `SecurityManagerListener`. This function is called with two parameters: the connection key, and a `HandshakeResult` enum value (one of Fail, Success, or CertExpired)
```

Comment:
The `SSLContext::HandshakeResult` has more values than the ones you listed here
```c++
enum HandshakeResult {
    Handshake_Result_Success,
    Handshake_Result_Fail,
    Handshake_Result_AbnormalFail,
    Handshake_Result_CertExpired,
    Handshake_Result_NotYetValid,
    Handshake_Result_CertNotSigned,
    Handshake_Result_AppIDMismatch,
    Handshake_Result_AppNameMismatch,
  };
```
Maybe don't list them and instead link HanshakeResult to this enum
```

Reply:
Only the values I mentioned are ever returned. I agree linking to the enum is a good idea. Done in https://github.com/smartdevicelink/sdl_core_guides/pull/78/commits/217669c12a6345fe479cc53e258a66a05c1b1a7d